drgn/cli.py

```diff
@@ -263,8 +264,10 @@ def _main() -> None:
             prog.set_core_dump(args.core)
         elif args.pid is not None:
             prog.set_pid(args.pid or os.getpid())
-        else:
+        elif os.geteuid() == 0:
             prog.set_kernel()
+        else:
+            prog.set_core_dump_fd(open_via_sudo("/proc/kcore", os.O_RDONLY))
```

Comment:
Again instead of checking the UID, how about:

```suggestion
        else:
            try:
                prog.set_kernel()
            except PermissionError:
                prog.set_core_dump_fd(open_via_sudo("/proc/kcore", os.O_RDONLY))
```